include/kframework/lemmas/lemmas.k

```diff
@@ -14,6 +14,23 @@ module LEMMAS [symbolic]
     imports EVM-INT-SIMPLIFICATION
     imports LEMMAS-HASKELL
 
+  // ########################
+  // Boolean
+  // ########################
+
+    rule B ==Bool true  => B         [simplification, comm]
+    rule B ==Bool false => notBool B [simplification, comm]
+
+    rule notBool notBool X => X [simplification]
+
+    rule { true  #Equals X     ==K   Y     } => { X #Equals Y } [simplification]
+    rule { true  #Equals X:Int ==Int Y:Int } => { X #Equals Y } [simplification]
+    rule { false #Equals X     ==K   Y     } => #Not ( { X #Equals Y } ) [simplification]
+    rule { false #Equals X:Int ==Int Y:Int } => #Not ( { X #Equals Y } ) [simplification]
```

Comment:
```suggestion
    rule { true  #Equals X ==K   Y } => { X #Equals Y } [simplification]
    rule { true  #Equals X ==Int Y } => { X #Equals Y } [simplification]
    rule { false #Equals @B } => #Not { true #Equals @B } [simplification]
```

Sorts are inferred automatically for `X` and `Y` because they are under the `_==Int_` symbol. Also, would adding one lemma to handle `{ false #Equals B }` help here?

Another thing to look at: https://kframework.org/k-distribution/include/kframework/builtin/domains/#implementation-of-booleans

Notice that we're using set variables there (prefixed with `@` instead).

Also notice, module `INT-KORE` seems to have many of these lemmas already? https://kframework.org/k-distribution/include/kframework/builtin/domains/#implementation-of-integers. Are we including them here instead of rewriting them?

Reply:
Aha, ok, excellent, I didn't notice them in `INT-SYMBOLIC`. I think we should probably have a list of them somewhere so that all of the relevant libraries are included. I'll go through and cross-check.

Reply:
@ehildenb I've looked at all of the `[...]-SYMBOLIC` modules, as well as the `lemmas.k` file, so now there should be no duplicates.